### ISSUE TITLE
NetscriptJS dynamic property access still allows RAM circumvention

### ISSUE THREAD

**Issue by ethankaminski** [danielyxie/bitburner#304]
Following up from #257, doing a dynamic access on `ns` properties still allows the player to reduce RAM cost.

The script below costs 1.95 GB RAM (omitting the cost of `weaken`/`grow`/`hack`), whereas the normal version would cost 2.35GB.

I'm not sure exactly how this bypass works - I was actually testing whether I could trick it into letting me pay the cost of only 1 of those 3 ns functions, but some how it's letting me pay the cost of 0 of them. If I also try to bypass `getServerSecurityLevel` and `getServerMoneyAvailable`, it DOES detect that fishy business is going on, so it doesn't allow infinitely-much trickery at least.

```js
export async function main(ns) {
    var server = ns.args[0]
    var mThres = ns.args[1]
    var secThres = ns.args[2]

    while (true) {
        if (ns.getServerSecurityLevel(server) > secThres) {
        	await ns['weaken'](server)
        	ns.tprint("one")
        } else if (ns.getServerMoneyAvailable(server) < mThres) {
        	await ns['grow'](server)
        	ns.tprint("two")
        } else {
        	await ns['hack'](server)
        	ns.tprint("three")
        }
    }
}
```

**Comment by hydroflame** [comment]
FYI I couldn't reproduce it in prod with that exact same script
```Script runtime error: 
Server Ip: 90.4.5.0
Script name: circumvent.ns
Args:[foodnstuff, 5625000, 3]
Dynamic RAM usage calculated to be greater than initial RAM usage. This is probably because you somehow circumvented the static RAM calculation.

Please don't do that :(```

**Comment by kopelli** [comment]
I don't know if I completely agree with outright removing the cost. From a computer science perspective, stack/register-based execution (which Netscript essentially seems to be is modeled on) still takes up some amount of RAM for storing the jump/break points in the stack. But I think the overall cost per each should be dropped significantly.

As it currently stands, the least expensive operations have a CONSTANTS cost of 0.05. I'd say flow control structures should be a fraction of that. Something like 0.005. Maybe even 0.001.

**Comment by danielyxie** [comment]
Should be fixed now in v0.39.0
